tls/s2n_resume.c

```diff
@@ -242,7 +239,25 @@ static S2N_RESULT s2n_tls13_client_deserialize_session_state(struct s2n_connecti
     return S2N_RESULT_OK;
 }
 
-static S2N_RESULT s2n_client_deserialize_session_state(struct s2n_connection *conn, struct s2n_stuffer *from)
+static S2N_RESULT s2n_deserialize_resumption_state(struct s2n_connection *conn, struct s2n_blob *identity, struct s2n_stuffer *from)
+{
+    RESULT_ENSURE_REF(conn);
```

Comment:
This method is almost identical to `s2n_client_deserialize_session_state`. Maybe you should just have one version of this method, and check the mode when choosing the how to deserialize S2N_TLS12_SERIALIZED_FORMAT_VERSION?